File: taller_03/nodejs/ejercicio.js

function suma(a, b){
    return a + b;
}


function esPar(n) {
    return n % 2 === 0;
}

describe('Test de esPar', () => {
    const testCases = [
        {input: 2, expected: true},
        {input: 0, expected: true},
        {input: -2, expected: true},
        {input: 1, expected: false}
    ];

    testCases.forEach(testCase =>  {
        it(`Debería retornar ${testCase.expected} para n = ${testCase.input}`, () => {
            const result = esPar(testCase.input);
            expect(result).toBe(testCase.expected);
        });
    });
});


describe('Test de suma', () => {
    const testCases = [
        {a: 1, b: 2, expected: 3},
        {a: 0, b: 0, expected: 0},
        {a: -1, b: 1, expected: 0},
        {a: 100, b: 200, expected: 300}
    ];

    testCases.forEach(testCase => {
        it(`Debería retornar ${testCase.expected} para a = ${testCase.a} y b = ${testCase.b}`, () => {
            const result = suma(testCase.a, testCase.b);
            expect(result).toBe(testCase.expected);
        });
    });
});


module.exports = { suma }

File: taller_03/python/ejercicio.py
import unittest

def suma(a,b):
    return a+b

def is_activate(user):
    return user.status



class TestSuma(unittest.TestCase):
    def test_suma(self):
        test_data = [
            (1, 2, 3),
            (0, 0, 0),
            (-1, 1, 0),
            (100, 200, 300)
        ]
        for data in test_data:
            result = suma(data[0], data[1])
            self.assertEqual(result, data[2])

#TODO Es par



def is_even(n):
    return n % 2  == 0


def test_is_even():
    test_cases = [
        (2, True),
        (0, True),
        (-2, True),
        (1, False)
    ]

    for n, expected in test_cases:
        result = is_even(n)
        assert result == expected, f"Para n = {n}, se esperaba {expected} pero se obtuvo {result}"



if __name__ == '__main__':
   test_is_even()